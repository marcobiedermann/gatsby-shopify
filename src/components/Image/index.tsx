import { GatsbyImage } from 'gatsby-plugin-image';
import React, { FC } from 'react';

interface Source {
  srcSet: string;
  sizes: string;
  type: string;
}

type Layout = 'fixed' | 'fullWidth' | 'constrained';

export interface ImageProps {
  altText: string;
  className?: string;
  gatsbyImageData: {
    images: {
      sources: Source[];
      fallback: {
        src: string;
        srcSet: string;
        sizes: string;
      };
    };
    layout: Layout;
    width: number;
    height: number;
  };
  id?: string;
}

const Image: FC<ImageProps> = (props) => {
  const { altText, gatsbyImageData, ...otherProps } = props;

  return <GatsbyImage image={gatsbyImageData} alt={altText} {...otherProps} />;
};

export default Image;
