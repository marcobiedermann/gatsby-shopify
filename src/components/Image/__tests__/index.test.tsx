import { render } from '@testing-library/react';
import React from 'react';
import Image from '..';

describe('image component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Image />);

    expect(asFragment()).toMatchSnapshot();
  });
});
