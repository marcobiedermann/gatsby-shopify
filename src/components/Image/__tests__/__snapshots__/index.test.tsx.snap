// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image component should render correctly 1`] = `
<DocumentFragment>
  <img />
</DocumentFragment>
`;
